src/Sylius/Bundle/CoreBundle/spec/Validator/Constraints/ImageUniqueCodeValidatorSpec.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\CoreBundle\Validator\Constraint;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\CoreBundle\Validator\Constraints\ImageUniqueCode;
+use Sylius\Bundle\CoreBundle\Validator\Constraints\ImageUniqueCodeValidator;
+use Sylius\Component\Core\Model\ImageAwareInterface;
+use Sylius\Component\Core\Model\ImageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * @mixin ImageUniqueCodeValidator
+ *
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class ImageUniqueCodeValidatorSpec extends ObjectBehavior
+{
+    function let(RepositoryInterface $imageRepository)
+    {
+        $this->beConstructedWith($imageRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ImageUniqueCodeValidator::class);
+    }
+    
+    function it_extends_constraint_validator()
+    {
+        $this->shouldHaveType(ConstraintValidator::class);
+    }
+    
+    function it_adds_violation_if_there_two_images_with_the_same_owner_which_have_same_codes(
+        ImageInterface $firstImage,
+        ImageInterface $secondImage,
+        ImageInterface $thirdImage,
+        ImageAwareInterface $owner,
+        ImageUniqueCode $constraint,
```

Comment:
And ok, it should be create by new. But not because that the message is null, but more because the kind of instance of ImageUnigeCode does not metter.